pages/Identity/Registry.md

```diff
@@ -0,0 +1,173 @@
+     * MUST be called by someone who is authorized on the contract via `EIP 1271(addr).isValidSignature`
+     * MUST emit DSNPHandle
+     */
+    function register(address addr, string handle) external;
+
+    /**
+     * @dev Register a new handle By EIP-712 Signature
+     * @param r ECDSA Signature r value
+     * @param s ECDSA Signature s value
+     * @param v EIP-155 calculated Signature v value
+     * @param addr Address for the handle to point at
+     * @param handle The handle for the address
+     * 
+     * MUST NOT allow a registration of an addr that is already attached to a different handle
+     * MUST be signed by someone who is authorized on the contract via `EIP 1271(addr).isValidSignature`
+     * MUST emit DSNPHandle
+     */
+    function registerByEIP712Sig(bytes32 r, bytes32 s, uint32 v, address addr, string handle) external;
+
+    /**
+     * @dev Alter a handle resolution
+     * @param newAddr New address for the handle to point at
+     * @param handle The handle to alter
+     * 
+     * MUST NOT allow a registration of an newAddr that is already attached to a different handle
+     * MUST be called by someone who is authorized on the contract via `EIP 1271(oldAddr).isValidSignature`
+     * TODO: FIX THE ISSUE OF newAddr not being a part of the creation
+     * MUST emit DSNPHandleOwnershipChange
+     */
+    function setAddr(address newAddr, string handle) external;
+
+    /**
+     * @dev Register a new handle By EIP-712 Signature
+     * @param r ECDSA Signature r value
+     * @param s ECDSA Signature s value
+     * @param v EIP-155 calculated Signature v value
+     * @param newAddr New address for the handle to point at
+     * @param handle The handle to alter
+     * 
+     * MUST NOT allow a registration of an addr that is already attached to a different handle
+     * MUST be signed by someone who is authorized on the contract via `EIP 1271(oldAddr).isValidSignature`
+     * TODO: FIX THE ISSUE OF newAddr not being a part of the creation
+     * MUST emit DSNPHandle
+     */
+    function setAddrSignature(bytes32 r, bytes32 s, uint32 v, address newAddr, string handle) external;
+
+    /**
+     * @dev Resolve a handle to a contract address
+     * @param handle The handle to resolve
+     * 
+     * @throws if not found
+     * @returns Address of the contract
+     */
+    function resolveHandleToAddress(string handle) view external returns (address);
+
+
+    /**
+     * @dev Resolve a handle to the DSNP Id
+     * @param handle The handle to resolve
+     * 
+     * @throws if not found
+     * @returns The uint64 DSNP Id for the handle
+     */
+    function resolveHandleToId(string handle) view external returns (uint64);
+
+    /**
+     * @dev Resolve a DSNP Id to a contract address
+     * @param id The DSNP Id to resolve
+     * 
+     * @throws if not found
+     * @returns Address of the contract
+     */
+    function resolveIdToAddress(uint64 id) view external returns (address);
```

Comment:
We don't have a `resolveAddressToId` for reverse lookup, but do we want that?

Reply:
Added a lot of details on how to do lookups in any which way someone might want.